backend/app/Actions/Visitors/GetBounceRateAction.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Visitors;
+
+use App\Repositories\Contracts\VisitorRepository;
+
+final class GetBounceRateAction
+{
+    private $visitorRepository;
+
+    public function __construct(VisitorRepository $visitorRepository)
+    {
+        $this->visitorRepository = $visitorRepository;
+    }
+
+    public function execute(): GetBounceRateResponse
+    {
+        $visitors = $this->visitorRepository->withSinglePageInactiveSession()->count();
```

Comment:
count is the aggregate function of a database.

And you should return data for a chart. It means it is not a single value, it should be a collection of bounce rates for a chosen date range.

Reply:
You should create some GetBounceRateRequest that has some filters, like startDate and endDate.

Reply:
And also some period: hour, day, week, month.

And according to these filters you should retrieve visitors that have only one finished session with one visit. And calculate bounce rate for each period